pkg/ccl/changefeedccl/avro_test.go

```diff
@@ -343,3 +381,50 @@ func TestAvroMigration(t *testing.T) {
 		})
 	}
 }
+
+func TestDecimalRatRoundtrip(t *testing.T) {
+	t.Run(`table`, func(t *testing.T) {
+		tests := []struct {
+			scale int32
+			dec   *apd.Decimal
+		}{
+			{0, apd.New(0, 0)},
+			{0, apd.New(1, 0)},
+			{0, apd.New(-1, 0)},
+			{0, apd.New(123, 0)},
+			{0, apd.New(1, 1)},
+			{0, apd.New(1, 3)},
+			{1, apd.New(0, -1)},
+			{1, apd.New(1, -1)},
+			{1, apd.New(123, -1)},
+			{5, apd.New(1, -5)},
+		}
+		for d, test := range tests {
+			rat, err := decimalToRat(*test.dec, test.scale)
+			require.NoError(t, err)
+			roundtrip := ratToDecimal(rat, test.scale)
+			if test.dec.Cmp(&roundtrip) != 0 {
```

Comment:
You may want to use `CmpTotal` here which does exact comparison and will catch differences with trailing zeros.